### PR TITLE
allow hand-sorting plugins

### PR DESCRIPTION
**Pull request by phryneas**
This would solve #226.

As I was almost done with that, @Tormak9970 mentioned that they had their own implementation to list sorting, so maybe that might be preferrable over this - but either way, maybe some other code from this PR might be salvageable :)

[Screenshot of the orderable list by Tormak9970](https://github.com/Tormak9970/portfolio-site-v2/blob/main/public/img/projs/Bash%20Shortcuts%20-%20Manage%20View.png)

[Screenshot of my solution](https://steamcommunity.com/sharedfiles/filedetails/?id=2900472013)

I went for the approach of having a sorted array of plugin names and keeping that in a setting.  
This required me to add an `EventTarget` to the `useSettings` hook so that updating a setting in one screen will synchronize to all other screens. Granted, this could also be done with an additional Context, but seems to work very nicely and imitates a pattern that is already used elsewhere in Decky.

### CODE REVIEW

**Comment by JediRhymeTrix** [comment]
Quick question, does this prevent newly updated plugins from always migrating to the bottom?

**Review by AAGaming00 — CHANGES_REQUESTED** [review]

File: frontend/src/utils/hooks/useSetting.ts
Comment (phryneas) [reply frontend/src/utils/hooks/useSetting.ts:19]: It is an EventListener for all events of `type` "`key`"  (see `settingUpdates.addEventListener(key, listener)` )  below. So each setting has their own event type and they are separated that way.

**Comment by phryneas** [comment]
> Quick question, does this prevent newly updated plugins from always migrating to the bottom?

New plugins land at the bottom and then you can sort them to a specific position, where they will stay. They have to start somewhere though.

File: frontend/src/utils/hooks/useSetting.ts
Comment (AAGaming00) [reply frontend/src/utils/hooks/useSetting.ts:19]: Ah

**Comment by AAGaming00** [comment]
On hold until Tormak's reorderable component is merged.

**Comment by TrainDoctor** [comment]
Closed in favor of an upcoming solution being PRed into DFL.

**Pull request closed without merging**
